### ISSUE TITLE
Can't find shimx64.efi.signed when building Ubuntu 20.04(.6) image

### ISSUE THREAD

**Issue by jloeser** [OSInside/kiwi#2275]
## Problem description

One month ago I was able to build Ubuntu 20.04 Focal ISO image using `flags="dmsquash"` and `firmware="uefi"`.

Today I started new build with same Kiwi version and same image description and it failed with:

```
[ WARNING ]: 15:53:48 | --> Running fallback setup for shim secure boot efi image
[ DEBUG   ]: 15:53:48 | EXEC: [cp /tmp/ubuntu/build/image-root/usr/lib/shim/shimx64.efi.signed /tmp/ubuntu/live-media.qha11vdi/EFI/BOOT/bootx64.efi]
[ DEBUG   ]: 15:53:48 | EXEC: Failed with stderr: cp: cannot stat '/tmp/ubuntu/build/image-root/usr/lib/shim/shimx64.efi.signed': No such file or directory
, stdout: (no output on stdout)
[ ERROR   ]: 15:53:48 | KiwiCommandError: cp: stderr: cp: cannot stat '/tmp/ubuntu/build/image-root/usr/lib/shim/shimx64.efi.signed': No such file or directory
, stdout: (no output on stdout)
```

## Steps to reproduce the behaviour

Change the image description (https://github.com/OSInside/kiwi-descriptions/blob/master/ubuntu/x86_64/ubuntu-focal/config.xml)
```
diff --git a/ubuntu/x86_64/ubuntu-focal/config.xml b/ubuntu/x86_64/ubuntu-focal/config.xml
index 099a4df..aab10a1 100644
--- a/ubuntu/x86_64/ubuntu-focal/config.xml
+++ b/ubuntu/x86_64/ubuntu-focal/config.xml
@@ -22,7 +22,7 @@
         <rpm-check-signatures>false</rpm-check-signatures>
     </preferences>
     <preferences profiles="Live">
-        <type image="iso" flags="overlay" hybridpersistent_filesystem="ext4" hybridpersistent="true" firmware="efi"/>
+        <type image="iso" flags="dmsquash" hybridpersistent_filesystem="ext4" hybridpersistent="true" firmware="uefi"/>
     </preferences>
     <preferences profiles="Virtual">
         <type image="vmx" filesystem="ext4" kernelcmdline="console=ttyS0" firmware="uefi" bootloader="grub2"/>
```

## OS and Software information

* KIWI version: 9.24.58
* Operating system host version: openSUSE Leap 15.4
* Operating system target version: Ubuntu 20.04 Focal using HWE kernel
* Open Build Service version (N/A if not using OBS): 
* Koji version (N/A if not using Koji): 

---

Looks like Ubuntu has removed `shim*.efi` from its latest `shim-signed_1.40.9+15.7-0ubuntu1_amd64.deb` package. The following change worked for me (see commit message for more information):

https://github.com/OSInside/kiwi/pull/2274

**Comment by jloeser** [comment]
Not quite sure what Ubuntu is doing there, maybe they are in some transition phase of exchanging vendor certificates in shim?

Also no idea what's the actual difference between `shimx64.efi.signed.latest` and `shimx64.efi.signed.previous`. Either M$ signatures have changed or (as just said) the vendor certificates inside shim binary. I couldn't find any information on this at the moment.

So maybe this fix is only temporary needed until Ubuntu starts shipping again "normal" `shim*.efi.signed` binary.

**Comment by schaefi** [comment]
> Not quite sure what Ubuntu is doing there, maybe they are in some transition phase of exchanging vendor certificates in shim?

I was also questioning myself about it but I think you are right there are some changes on how secure boot is handled and I also have no insights. As long as the file names for the EFI binaries are on flux I fear kiwi will be in trouble though.

**Comment by jloeser** [comment]
Some more information: https://lists.ubuntu.com/archives/jammy-changes/2023-February/018925.html

```
...
  * Import 20221103 Canonical vendor dbx.
    This vendor dbx revokes all certificates that have been used
    so far.
    - CN = Canonical Ltd. Secure Boot Signing
    - CN = Canonical Ltd. Secure Boot Signing (2017)
    - CN = Canonical Ltd. Secure Boot Signing (ESM 2018)
    - CN = Canonical Ltd. Secure Boot Signing (2019)
    - CN = Canonical Ltd. Secure Boot Signing (Ubuntu Core 2019)
    - CN = Canonical Ltd. Secure Boot Signing (2021 v1)
    - CN = Canonical Ltd. Secure Boot Signing (2021 v2)
    - CN = Canonical Ltd. Secure Boot Signing (2021 v3)
...
```

**Comment by schaefi** [comment]
Fixed with #2274
